File: src/main/java/com/gmail/borlandlp/minigamesdtools/gui/inventory/api/ViewManager.java
package com.gmail.borlandlp.minigamesdtools.gui.inventory.api;

import com.gmail.borlandlp.minigamesdtools.APIComponent;
import com.gmail.borlandlp.minigamesdtools.gui.inventory.DrawableInventory;
import com.gmail.borlandlp.minigamesdtools.gui.inventory.api.InventoryAPI;
import org.bukkit.entity.Player;

import java.util.Hashtable;
import java.util.Map;

public class ViewManager implements InventoryAPI, APIComponent {
    private Map<String, DrawableInventory> viewMap = new Hashtable<>();

    public void show(DrawableInventory drawableInventory, Player player) {
        if(player.getOpenInventory() != null) {
            player.getOpenInventory().close();
        }

        this.viewMap.put(player.getName(), drawableInventory);
        player.openInventory(drawableInventory.toBukkitInventory());
    }

    public DrawableInventory getInventory(Player player) {
        return this.viewMap.get(player.getName());
    }

    public boolean isView(Player player) {
        return this.viewMap.containsKey(player.getName());
    }

    public void close(Player player) {
        this.unregister(player);
    }

    public void unregister(Player player) {
        this.viewMap.remove(player.getName());
    }

    @Override
    public void onLoad() {

    }

    @Override
    public void onUnload() {

    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/gui/inventory/api/ViewManager.kt
package com.gmail.borlandlp.minigamesdtools.gui.inventory.api

import com.gmail.borlandlp.minigamesdtools.APIComponent
import com.gmail.borlandlp.minigamesdtools.gui.inventory.DrawableInventory
import org.bukkit.entity.Player
import java.util.*

class ViewManager : InventoryAPI, APIComponent {
    private val viewMap: MutableMap<String, DrawableInventory> =
        Hashtable()

    override fun show(drawableInventory: DrawableInventory, player: Player) {
        player.openInventory?.close()
        viewMap[player.name] = drawableInventory
        player.openInventory(drawableInventory.toBukkitInventory())
    }

    override fun getInventory(player: Player): DrawableInventory {
        return viewMap[player.name]!!
    }

    override fun isView(player: Player): Boolean {
        return viewMap.containsKey(player.name)
    }

    override fun close(player: Player) {
        unregister(player)
    }

    override fun unregister(player: Player) {
        viewMap.remove(player.name)
    }

    override fun onLoad() {}
    override fun onUnload() {}
}